### PR TITLE
[feature] ROS plugins v2 out of source tree builds

### PR DESCRIPTION
**Pull request by artivis**
This PR changes the ROS plugins v2 so that they use the 'out of source tree builds' feature introduced in https://github.com/snapcore/snapcraft/pull/3392.

It allows for distributing the `snapcraft.yaml` config file directly alongside a ROS package (in the package root directory) whereas it can currently only be distributed with a ROS workspace.

Replaces a previous attempt https://github.com/snapcore/snapcraft/pull/3348.

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [X] Have you successfully run `./runtests.sh static`?
- [X] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Review by sergiusens — APPROVED** [review]

File: tests/spread/plugins/v2/snaps/catkin-tools-noetic-in-source-hello/package.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>snapcraft_hello</name>
+  <version>0.0.1</version>
+  <description>snapcraft test for roscpp</description>
+  <maintainer email="me@example.com">me</maintainer>
+  <license>Apache License 2.0</license>
```
Comment (cjp256) [inline tests/spread/plugins/v2/snaps/catkin-tools-noetic-in-source-hello/package.xml:7]: I know this was probably copy/paste, but the license field here doesn't match the source file headers.  Maybe should be gpl3?

**Review by cjp256 — APPROVED** [review]
LGTM, just the nit on the license field.

File: tests/spread/plugins/v2/snaps/catkin-tools-noetic-in-source-hello/package.xml (hunk 1 above)
Comment (artivis) [reply tests/spread/plugins/v2/snaps/catkin-tools-noetic-in-source-hello/package.xml:7]: Indeed this was copy/pasted from the [catkin-noetic-hello](https://github.com/snapcore/snapcraft/blob/1df75d9d6a032523a2fa1ca06e634e50e4574dcf/tests/spread/plugins/v2/snaps/catkin-noetic-hello/src/snapcraft_hello/package.xml) test (and duplicated in [catkin-noetic-in-source-hello](https://github.com/snapcore/snapcraft/blob/1df75d9d6a032523a2fa1ca06e634e50e4574dcf/tests/spread/plugins/v2/snaps/catkin-noetic-in-source-hello/package.xml) test too). Should I fix them all?

Comment (sergiusens) [reply tests/spread/plugins/v2/snaps/catkin-tools-noetic-in-source-hello/package.xml:7]: A follow up PR fixing them all is fine

**Pull request merged**